x/group/keeper/msg_server.go

```diff
@@ -345,12 +345,14 @@ func (k Keeper) CreateGroupPolicy(goCtx context.Context, req *group.MsgCreateGro
 			// next sequence value and derive a new address.
 			continue
 		}
-		acc := k.accKeeper.NewAccount(ctx, &authtypes.ModuleAccount{
-			BaseAccount: &authtypes.BaseAccount{
-				Address: accountAddr.String(),
-			},
-			Name: accountAddr.String(),
-		})
+
+		// group policy accounts are base accounts
+		baseAccount := authtypes.NewBaseAccountWithAddress(accountAddr)
+		if err := baseAccount.Validate(); err != nil {
```

Comment:
I think the idea discussed in the Slack thread was to do:

```
acct := NewGroupPolicyInfo()
```

where in another file we do:

```
var _ authtypes.AccountI = &GroupPolicyInfo{}

func (a *GroupPolicyInfo) Validate() {
  // Correctly validate addresses using ADR-028
}
```

Reply:
If we do this, we also need a bigger migration, i.e. populate the `derivation_key` field in the GroupPolicyInfo proto file, like in 452f281518eddbd67d52c040830aee1362a6ff2f

Reply:
This means the addresses will change, right?